sdk/core/Azure.Core/src/Pipeline/Internal/ReadTimeoutStream.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Pipeline
+{
+    internal class ReadTimeoutStream : Stream
```

Comment:
Shouldn't we derive it from `ReadOnlyStream`?

Reply:
Is it a BCL type?

Reply:
It is our type, `Azure.Core.Pipeline.ReadOnlyStream`